### PR TITLE
feat: Omit local ID fields by default

### PR DESCRIPTION
**Pull request by taion**
Allow including them, possibly with deprecation, as non-default options.

### CODE REVIEW

**Comment by taion** [comment]
the default handling is to do that – but i thought the easiest way forward (here and elsewhere) as to add the ability to deprecate. i didn't want to spend the time to figure out a way to hook into the creation of deprecated local ID fields elsewhere. next step (once everything is migrated over) will be to remove local ID handling here entirely, though.

File: src/types/NodeType.js [hunk 1]
```diff
@@ -184,9 +186,14 @@ export default function createNodeType({
         id: globalIdField(null, getLocalId),
       };
 
+      // This will only be set if localIdFieldMode is not 'omit'.
       if (localIdFieldName) {
         idFields[localIdFieldName] = {
           type: GraphQLString,
+          deprecationReason:
+            localIdFieldMode === 'deprecated'
+              ? 'local IDs are deprecated'
```
Comment (jquense) [inline src/types/NodeType.js]: maybe mention the alternative?

File: src/types/NodeType.js [hunk 2]
```diff
@@ -233,8 +240,15 @@ export default function createNodeType({
...
       ...config
     }: NodeTypeConfig<R>) {
-      // eslint-disable-next-line no-param-reassign
-      localIdFieldName = getLocalIdFieldName(name, localIdFieldName);
+      if (localIdFieldMode !== 'omit') {
+        // eslint-disable-next-line no-param-reassign
+        localIdFieldName = getLocalIdFieldName(name, localIdFieldName);
+      } else {
+        invariant(
+          !localIdFieldName,
```
Comment (jquense) [inline src/types/NodeType.js]: this is fine because `makeId` actually is the thing that produces the local id for the global id resolver?

Comment (taion) [reply src/types/NodeType.js]: yup... these are just schemas so none of the fields interact (or can interact)

**Pull request merged**